CMake error, if version less than 3.8

If I want to create the json_dto library with a CMake version lesser than 3.8 I get the following error:

CMake Error at json_dto/CMakeLists.txt:13 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "CXX_STANDARD" is not allowed.

CMake Error at json_dto/CMakeLists.txt:13 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "CXX_STANDARD_REQUIRED" is not allowed.

CMake Error at json_dto/CMakeLists.txt:13 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "CXX_EXTENSIONS" is not allowed.

--- a/dev/json_dto/CMakeLists.txt
+++ b/dev/json_dto/CMakeLists.txt
@@ -10,11 +10,9 @@ add_library(${JSON_DTO} INTERFACE)
 add_library(json-dto::json-dto ALIAS ${JSON_DTO})
 
 IF (${CMAKE_VERSION} VERSION_LESS "3.8.0")
-	set_target_properties(${JSON_DTO} PROPERTIES
-		CXX_STANDARD 14
-		CXX_STANDARD_REQUIRED YES
-		CXX_EXTENSIONS NO
-	)
+	set(CMAKE_CXX_STANDARD 14)
+	set(CMAKE_CXX_STANDARD_REQUIRED ON)
+	set(CMAKE_CXX_EXTENSIONS OFF)
 ELSE ()
 	target_compile_features(${JSON_DTO} INTERFACE cxx_std_14)
 ENDIF ()

